From castet.matthieu@free.fr Tue Jan 17 22:37:55 2006
Message-ID: <43CDE29F.5080902@free.fr>
Date: Wed, 18 Jan 2006 07:39:27 +0100
From: matthieu castet <castet.matthieu@free.fr>
To: greg@kroah.com
Subject: [PATCH 3/2] UEAGLE : cmv name bug (was cosmetic)

this patch correct a possible bug with cmv_name being static. If there
is 2 modems and the driver is scheduled when filling cmv_name this could
result with garbage in cmv_name. We allocate cmv_name on the stack but
with a small size in order to avoid that.


Signed-off-by: Matthieu CASTET <castet.matthieu@free.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/atm/ueagle-atm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/atm/ueagle-atm.c
+++ gregkh-2.6/drivers/usb/atm/ueagle-atm.c
@@ -1013,7 +1013,7 @@ static int request_cmvs(struct uea_softc
 	int ret, size;
 	u8 *data;
 	char *file;
-	static char cmv_name[256] = FW_DIR;
+	char cmv_name[FIRMWARE_NAME_MAX]; /* 30 bytes stack variable */
 
 	if (cmv_file[sc->modem_index] == NULL) {
 		if (UEA_CHIP_VERSION(sc) == ADI930)
